source/tests/unit/xnet_converters_tests.cpp

```diff
@@ -435,6 +469,47 @@ TEST(XnetConvertersTests, LingerSockOptData_ConvertToGrpc_ConvertsToSockOptDataW
   EXPECT_EQ(L_ONOFF, grpc_data.data_linger().l_onoff());
 }
 
+TEST(XnetConvertersTests, IPMReqSockOptData_ConvertToGrpc_ConvertsToSockOptDataWithIPMReqValue)
+{
+  constexpr auto IMR_MULTIADDR = 22;
+  constexpr auto IMR_INTERFACE = 1;
+  auto storage = allocate_output_storage<void*, SockOptData>(OptName::OPT_NAME_IP_ADD_MEMBERSHIP);
+  void* data_pointer = storage.data();
+  EXPECT_EQ(data_pointer, &(storage.data_ipmreq));
+  auto ipmreq_pointer = reinterpret_cast<nxip_mreq*>(data_pointer);
+  ipmreq_pointer->imr_multiaddr.addr = IMR_MULTIADDR;
+  ipmreq_pointer->imr_interface.addr = IMR_INTERFACE;
+
+  auto grpc_data = SockOptData{};
+  convert_to_grpc(storage, &grpc_data);
+
+  EXPECT_EQ(SockOptData::DataCase::kDataIpmreq, grpc_data.data_case());
+  EXPECT_EQ(IMR_MULTIADDR, grpc_data.data_ipmreq().imr_multiaddr());
+  EXPECT_EQ(IMR_INTERFACE, grpc_data.data_ipmreq().imr_interface());
+}
+
+TEST(XnetConvertersTests, IPV6MReqSockOptData_ConvertToGrpc_ConvertsToSockOptDataWithIPMReqValue)
+{
+  const auto IPV6MR_MULTIADDR = std::vector<char>{0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xA, 0xB, 0xC, 0xD, 0xE, 0xF};
+  constexpr auto IPV6MR_INTERFACE = 1;
+  auto storage = allocate_output_storage<void*, SockOptData>(OptName::OPT_NAME_IPV6_ADD_MEMBERSHIP);
+  void* data_pointer = storage.data();
+  EXPECT_EQ(data_pointer, &(storage.data_ipv6mreq));
+  auto ipv6mreq_pointer = reinterpret_cast<nxipv6_mreq*>(data_pointer);
+  std::memcpy(
+      ipv6mreq_pointer->ipv6mr_multiaddr.addr,
+      IPV6MR_MULTIADDR.data(),
+      sizeof(ipv6mreq_pointer->ipv6mr_multiaddr.addr));
+  ipv6mreq_pointer->ipv6mr_interface = IPV6MR_INTERFACE;
+
+  auto grpc_data = SockOptData{};
+  convert_to_grpc(storage, &grpc_data);
+
+  EXPECT_EQ(SockOptData::DataCase::kDataIpv6Mreq, grpc_data.data_case());
+  EXPECT_THAT(grpc_data.data_ipv6mreq().ipv6mr_multiaddr(), ElementsAreArray(IPV6MR_MULTIADDR.data(), IPV6MR_MULTIADDR.size()));
```

Comment:
Can you add an assert here matching my other comment? I think you can assert that the multiaddr().size() on the grpc data equals the `IPV6MR_MULTIADDR.size()`.

Reply:
I went ahead and still added an assert here for ensuring the sizes are the same before comparing the elements.